internal/argocd/cluster/cluster.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 The argocd-agent Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+
+	"github.com/argoproj/argo-cd/v2/common"
+	appv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	cacheutil "github.com/argoproj/argo-cd/v2/util/cache"
+	appstatecache "github.com/argoproj/argo-cd/v2/util/cache/appstate"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConnectionStatusConnected    = "Connected"
+	ConnectionStatusDisconnected = "Disconnected"
+)
```

Comment:
These should use the proper values from Argo CD, so that we get the small icons from the UI as a bonus, e.g.

![image](https://github.com/user-attachments/assets/f787181f-3319-4570-b3b2-5e9fd6e94eb8)

Versus:

![image](https://github.com/user-attachments/assets/db58a4a1-af04-44df-97b2-a78017bc3bfa)